compiler/aarch64/codegen/FPTreeEvaluator.cpp

```diff
@@ -90,11 +90,12 @@ OMR::ARM64::TreeEvaluator::fconstEvaluator(TR::Node *node, TR::CodeGenerator *cg
 
    union {
       float f;
-      int32_t i;
+      double d;
    } fvalue;
 
    fvalue.f = node->getFloat();
-   loadConstant32(cg, node, fvalue.i, tmpReg);
+   uint32_t imm = getFPImm(fvalue.d);
```

Comment:
Using the union here does not work as you expect.  Just use a type cast from `float` to `double` for passing the value to `getFPImm()`.